From 9721b63522ef4c2d64a0b623e280fb795fb2ee7a Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Thu, 13 Sep 2018 13:07:47 +0800
Subject: [PATCH 4605/5242] MLK-19565-1: ASoC: fsl_rpmsg_i2s: support rpmsg on
 imx8qm

commit  b9f8333303f2f871bc81d9929b8defa017128427 from
https://source.codeaurora.org/external/imx/linux-imx.git

On imx8qm mek, the cs42888 is connected with i2c in cm41 domain,
but wm8960 is connected with i2c1, which is not in m4 domain.
So we only need to eable rpmsg for cs42888.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit 9d2368aef40e4d107e4deee1a2c7e191c1afe644)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../devicetree/bindings/sound/fsl,rpmsg-i2s.txt    |    1 +
 sound/soc/fsl/fsl_rpmsg_i2s.c                      |    8 ++++++++
 2 files changed, 9 insertions(+)

diff --git a/Documentation/devicetree/bindings/sound/fsl,rpmsg-i2s.txt b/Documentation/devicetree/bindings/sound/fsl,rpmsg-i2s.txt
index 5f8c6cb..27de48e 100644
--- a/Documentation/devicetree/bindings/sound/fsl,rpmsg-i2s.txt
+++ b/Documentation/devicetree/bindings/sound/fsl,rpmsg-i2s.txt
@@ -8,6 +8,7 @@ Required properties:
 
   - compatible		: Compatible list, contains "fsl,imx7ulp-rpmsg-i2s".
 				"fsl,imx8mq-rpmsg-i2s", "fsl,imx8qxp-rpmsg-i2s"
+				"fsl,imx8qm-rpmsg-i2s"
 
   - fsl,audioindex      : This is an index indicating the audio device index in
 			  the M4 side.
diff --git a/sound/soc/fsl/fsl_rpmsg_i2s.c b/sound/soc/fsl/fsl_rpmsg_i2s.c
index b7ace88..2d45f51 100644
--- a/sound/soc/fsl/fsl_rpmsg_i2s.c
+++ b/sound/soc/fsl/fsl_rpmsg_i2s.c
@@ -108,6 +108,7 @@ static int i2s_send_message(struct i2s_rpmsg *msg,
 	{ .compatible = "fsl,imx7ulp-rpmsg-i2s"},
 	{ .compatible = "fsl,imx8mq-rpmsg-i2s"},
 	{ .compatible = "fsl,imx8qxp-rpmsg-i2s"},
+	{ .compatible = "fsl,imx8qm-rpmsg-i2s"},
 	{ /* sentinel */ }
 };
 MODULE_DEVICE_TABLE(of, fsl_rpmsg_i2s_ids);
@@ -191,6 +192,13 @@ static int fsl_rpmsg_i2s_probe(struct platform_device *pdev)
 	}
 
 	if (of_device_is_compatible(pdev->dev.of_node,
+				    "fsl,imx8qm-rpmsg-i2s")) {
+		rpmsg_i2s->codec_wm8960 = 0;
+		rpmsg_i2s->version = 1;
+		rpmsg_i2s->codec_cs42888 = 1 + (0 << 16);
+	}
+
+	if (of_device_is_compatible(pdev->dev.of_node,
 				    "fsl,imx8mq-rpmsg-i2s")) {
 		rpmsg_i2s->codec_wm8960 = 0;
 		rpmsg_i2s->version = 2;
-- 
1.7.9.5

